src/main/java/com/google/api/codegen/util/py/PythonTypeTable.java

```diff
@@ -82,19 +94,101 @@ public String getAndSaveNicknameFor(String fullName) {
 
   @Override
   public String getAndSaveNicknameFor(TypeName typeName) {
-    return dynamicTypeTable.getAndSaveNicknameFor(typeName);
+    return typeName.getAndSaveNicknameIn(this);
   }
 
   @Override
   public String getAndSaveNicknameFor(TypeAlias alias) {
-    return dynamicTypeTable.getAndSaveNicknameFor(alias);
+    if (!alias.needsImport()) {
+      return alias.getNickname();
+    }
+
+    String shortName = alias.getNickname().substring(0, alias.getNickname().indexOf("."));
+    String className = alias.getNickname().substring(alias.getNickname().indexOf(".") + 1);
+    String moduleName =
+        alias.getFullName().substring(0, alias.getFullName().length() - className.length() - 1);
+    if (usedShortNames.containsKey(shortName)) {
+      // Short name already there.
+      String oldModuleName = usedShortNames.get(shortName);
+      if (moduleName.equals(oldModuleName)) {
+        // New alias for existing module import, no clash
+        moduleImports.put(moduleName, alias);
```

Comment:
We need to reference all the aliases for a given module name in case we need to replace them with a disambiguated name in `updateOldImports`.